### PR TITLE
Bump python to 3.13 following core 2024.12

### PR DESCRIPTION
**Pull request by CoMPaTech**
## Summary by CodeRabbit

- **New Features**
	- Added support for Python 3.13 across various workflows and configurations.

- **Bug Fixes**
	- Updated pre-commit hooks to ensure compatibility with Python 3.13.

- **Documentation**
	- Changelog updated to reflect the new version (v1.6.1) and support for Python 3.13.

- **Chores**
	- Incremented project version in `pyproject.toml` and updated Python requirements.
	- Modified scripts to support multiple Python versions (3.12 and 3.13).

### CODE REVIEW

File: .github/workflows/verify.yml [hunk 1]
```diff
@@ -173,7 +173,7 @@ jobs:
     needs: commitcheck
     strategy:
       matrix:
-        python-version: ["3.12"]
+        python-version: ["3.13"]
```
[the remark this answers is not in the document]
Comment (CoMPaTech) [reply .github/workflows/verify.yml:176]: Good call, but HA-core moves towards 3.13 so there is no need to remain testing in 3.12

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -6,7 +6,7 @@ ci:
 
 default_language_version:
   # force all unspecified python hooks to run python3
-  python: python3.12
+  python: python3.13
```
[the remark this answers is not in the document]
Comment (CoMPaTech) [reply .pre-commit-config.yaml:9]: As far as I can see they all are python3.13 compliant, but thanks for noting

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## v1.6.1
```
[the remark this answers is not in the document]
Comment (CoMPaTech) [reply CHANGELOG.md:3]: `pyproject.toml` changed indicating compatibility with 3.13 - hence 1.60 would only be displayed as compatible with 3.12 - as such bumping the version

[the remark this answers is not in the document]
Comment (bouwew) [reply CHANGELOG.md:3]: Ah OK :)

**Review by bouwew — APPROVED** [review]
LGTM!

**Pull request merged**